src/classes/MisSubcountyMarketPrices.cls

```diff
@@ -0,0 +1,851 @@
+    	
+        }
+        
+        List<Market_Bushenyi__c> bushenyiSubmissions = [SELECT Name FROM Market_Bushenyi__c WHERE Date_of_capture__c = yesterday ];
+    
+        for (Market_Bushenyi__c bushenyiSubmission : bushenyiSubmissions){
+        	String marketName = [SELECT X3_0_Market__r.Name FROM Market_Bushenyi__c WHERE Name = : bushenyiSubmission.Name ].Name;
+        	computeBushenyiMaizePriceAverages(marketName);
+        	computeBushenyiCoffeePriceAverages(marketName);
+        	computeBushenyiCassavaPriceAverages(marketName);
+        	computeBushenyiKBeansPriceAverages(marketName);
+        	computeBushenyiSweetPotatoesPriceAverages(marketName);
+        	computeBushenyiBananaPriceAverage(marketName);
+        	computeBushenyiNambalePriceAverage(marketName);
+        	computeBushenyiMilletPriceAverage(marketName);
+    	
+        }
+        
+    }
+    
+    public static void computeMasindiMaizePriceAverages(String marketName) {
+        //computes average retail and wholesale prices per Kg of maize
+        AggregateResult[] maizeResult = [
+            SELECT 
+                AVG(A_Kg_of_Mazie_grain_at_retail_price__c)retail, 
+                AVG(A_Kg_of_Mazie_grain_at_wholesale_price__c)wholesale
+            FROM 
+                X3_0_Masindi_Maize__c 
+            WHERE 
+                X3_0_Market_Masindi__r.Date_of_capture__c = yesterday ];   
+                
+        Menu_Item__c maizeMenuItem = [
+            SELECT  
+                Label__c, 
+                Parent_Item__r.Label__c 
+            FROM 
+                Menu_item__c 
+            WHERE 
+                Parent_Item__r.Label__c = : marketName
+            AND 
+                Label__c='Maize'];
+                
+        if(String.valueOf(maizeResult[0].get('retail')) != null && String.valueOf(maizeResult[0].get('wholesale')) != null){
+            Integer averageRetailPrice = Integer.valueOf(maizeResult[0].get('retail'));
+            Integer averageWholesalePrice = Integer.valueOf(maizeResult[0].get('wholesale'));
+            
+            maizeMenuItem.Content__c = 'Average Retail Price '+averageRetailPrice+' Per Kg'+
+                '\N Average Wholesale Price '+ averageWholesalePrice+' Per Kg';
+            update maizeMenuItem;
+            //return averageRetailPrice + ' '+ averageWholesalePrice;
+        } 
+    }
+    
+    public static void computeMasindiSunflowerPriceAverages(String marketName) {
+        //computes average retail and wholesale prices per Kg of Sunflower
+        AggregateResult[] sunflowerResult = [
+            SELECT 
+                AVG(A_Kg_of_Sunflower_at_retail_price__c)retail, 
+                AVG(A_Kg_of_Sunflower_at_wholesale_price__c)wholesale
+            FROM 
+                X3_0_Masindi_Sunflower__c 
+            WHERE 
+                X3_0_Market_Masindi__r.Date_of_capture__c = yesterday ]; 
+                
+            Menu_Item__c sunflowerMenuItem = [
```

Comment:
This is living on the edge. If for some reason the menuitem doesn't exist anymore, processing will stop with an exception because the query return no results.